java/vector/src/test/java/org/apache/arrow/vector/TestValueVector.java

```diff
@@ -2642,6 +2642,26 @@ public void testEqualsWithIndexOutOfRange() {
     }
   }
 
+  @Test
+  public void testNullHashCode() {
```

Comment:
please separate this to a test per type, it is also missing a test for union types?

Reply:
Good suggestion. Thanks.
I have added separate test cases for FixedWidthVector, VariableWidthVector and UnionVector.